lib/pbio/src/orientation.c

```diff
@@ -77,4 +86,203 @@ void pbio_orientation_get_complementary_axis(uint8_t *index, int8_t *sign) {
+    if (err != PBIO_SUCCESS) {
+        return err;
+    }
+
+    pbio_orientation_imu_set_heading(0.0f);
+
+    return PBIO_SUCCESS;
+}
+
+/**
+ * Checks if the IMU is currently stationary.
+ *
+ * @return    True if it has been stationary for about a second, false if moving.
+ */
+bool pbio_orientation_imu_is_stationary(void) {
+    return pbdrv_imu_is_stationary(imu_dev);
+}
+
+/**
+ * Sets the thresholds that define when the hub is stationary. When the
+ * measurements are steadily below these levels, the orientation module
+ * automatically recalibrates.
+ *
+ * @param [in]  angular_velocity Angular velocity threshold in deg/s.
+ * @param [in]  acceleration     Acceleration threshold in mm/s^2
+ */
+void pbio_orientation_imu_set_stationary_thresholds(float angular_velocity, float acceleration) {
+    int16_t gyro_threshold = pbio_int_math_bind(angular_velocity / imu_config->gyro_scale, 1, INT16_MAX);
+    int16_t accl_threshold = pbio_int_math_bind(acceleration / imu_config->accel_scale, 1, INT16_MAX);
+    pbdrv_imu_set_stationary_thresholds(imu_dev, gyro_threshold, accl_threshold);
+}
+
+/**
+ * Reads the current IMU angular velocity in deg/s, compensated for offset.
+ *
+ * @param [out] values      The angular velocity vector.
+ */
+void pbio_orientation_imu_get_angular_velocity(pbio_geometry_xyz_t *values) {
+    pbio_geometry_vector_map(&pbio_orientation_base_orientation, &angular_velocity, values);
+}
+
+/**
+ * Reads the current IMU acceleration in mm/s^2.
+ *
+ * @param [out] values      The acceleration vector.
+ */
+void pbio_orientation_imu_get_acceleration(pbio_geometry_xyz_t *values) {
+    pbio_geometry_vector_map(&pbio_orientation_base_orientation, &acceleration, values);
+}
+
+/**
+ * Gets which side of a hub points upwards.
+ *
+ * @return                  Which side is up.
+ */
+pbio_orientation_side_t pbio_orientation_imu_get_up_side(void) {
+    // Up is which side of a unit box intersects the +Z vector first.
+    // So read +Z vector of the inertial frame, in the body frame.
+    // For now, this is the gravity vector. In the future, we can make this
+    // slightly more accurate by using the full IMU orientation.
+    return pbio_orientation_side_from_vector(&acceleration);
+}
+
+static float heading_offset = 0;
+
+/**
+ * Reads the estimated IMU heading in degrees, accounting for user offset.
+ *
+ * @return                  Heading angle in the base frame.
+ */
+float pbio_orientation_imu_get_heading(void) {
+
+    pbio_geometry_xyz_t heading_mapped;
+
+    pbio_geometry_vector_map(&pbio_orientation_base_orientation, &heading, &heading_mapped);
+
+    return heading_mapped.z - heading_offset;
+}
+
+/**
+ * Resets the IMU heading.
+ *
+ * This only adjusts the user offset without resetting anything in the
+ * algorithm, so this can be called at any time.
+ *
+ * @param [in] desired_heading  The desired heading value.
+ */
+void pbio_orientation_imu_set_heading(float desired_heading) {
```

Comment:
```suggestion
void pbio_orientation_imu_reset_heading(float desired_heading) {
```

To match description.